osprey_flask_app/routes.py

```diff
@@ -44,17 +46,51 @@ def osprey_route():
     """
     args = request.args
     arg_dict = create_full_arg_dict(args)
-    outpath = run_full_rvic(arg_dict)
+    validate_response = validate_inputs(arg_dict)
+    if validate_response.status_code == 400:
+        return validate_response
+
+    rvic_thread = threading.Thread(
+        target=lambda q, arg: q.put(run_full_rvic(arg)), args=(que, arg_dict)
+    )
+    rvic_thread.start()
+    return Response(
+        "RVIC Process started. Check status: "
+        + url_for("osprey.status_route", thread_id=rvic_thread.native_id),
+        status=202,
+    )
+
+
+@osprey.route("/status/<thread_id>", methods=["GET"])
+def status_route(thread_id):
+    """Provide route to check status of RVIC process."""
+    active_thread_ids = [str(t.native_id) for t in threading.enumerate()]
+    if thread_id in active_thread_ids:
+        return Response("Process is still running.", status=201)
+    else:
+        return Response(
+            "Process completed. Get output: "
+            + url_for("osprey.output_route", thread_id=thread_id),
+            status=200,
+        )
+
+
+@osprey.route("/output/<thread_id>", methods=["GET"])
+def output_route(thread_id):
+    """Provide route to get streamflow output of RVIC process."""
+    if que.empty():
+        return Response("Process has failed. No output returned.", status=404)
```

Comment:
Is this necessarily true? What if the user sent you a random job id? Then the job didn't fail, it just never existed. 404 is the appropriate status, but it would be good to differentiate between a failed job and a non-existent job.